modules/ublog/src/main/UblogSearch.scala

```diff
@@ -0,0 +1,24 @@
+package lila.ublog
+
+import lila.core.id.UblogPostId
+import lila.search.*
+import lila.search.client.SearchClient
+import lila.search.spec.{ Query, SortBlogsBy }
+import UblogAutomod.Quality
+
+final class UblogSearch(client: SearchClient, config: UblogConfig)(using Executor)
+    extends SearchReadApi[UblogPostId, Query.Ublog]:
+  lazy val builder = PaginatorBuilder(this, config.searchPageSize)
+
+  def fetchResults(text: String, by: lila.core.ublog.BlogsBy, minQualityOpt: Option[Quality], page: Int) =
+    val sortBy =
+      SortBlogsBy.values.find(_.toString == by.toString).getOrElse(SortBlogsBy.Score)
```

Comment:
We should atleast map one type to another using pattern match. Using `toString` for mapping is dangerous, we can update type and nobody knows it is broken.